fs/path.go

```diff
@@ -122,6 +122,8 @@ func sameFile(f1, f2 *currentPath) (bool, error) {
 				eq, err = compareSymlinkTarget(f1.fullPath, f2.fullPath)
 			} else if f1.f.Size() > 0 {
 				eq, err = compareFileContent(f1.fullPath, f2.fullPath)
+			} else {
+				eq, err = true, nil
```

Comment:
nit.. no need to set err=nil.. it's already nil

Reply:
True. I just did this because it had nice symmetry with the lines above. Happy to change it to `eq = true` if you would prefer.

Reply:
hmm could start out with eq := true  ..

less else more symmetry

Reply:
That feels kind of tricky. The else makes it clear that you expect to fall through and you're handling that case. Personally, I'd prefer the else with the `eq = true` to the "implicit fall through." It would make it easier for future readers to see the intent.

Reply:
if you want less tricky implicit fall throughs 
```
if (f1.f.Mode() & os.ModeSymlink) == os.ModeSymlink {
   eq, err := compareSymlinkTarget(f1.fullPath, f2.fullPath)
   if err != nil || !eq {
      return eq, err
   }
} else if f1.f.Size() > 0 {
   eq, err := compareFileContent(f1.fullPath, f2.fullPath)
   if err != nil || !eq {
      return eq, err
   }
} 
return true, nil // files match || f1.f.Size() == 0
```

Reply:
even better you could use a switch block

Reply:
Even in golang, there are a lot of ways to write the same logic. TBH, if were up to me, I'd stick with the first way I wrote it which is a minimal change and very clear to the reader. But I'm happy to go along with whatever the maintainers prefer.

Reply:
add a comment for your additional "else" case

Reply:
Good idea. Done.